core/primitives/src/runtime/parameter_table.rs

```diff
@@ -156,89 +189,105 @@ impl ParameterTable {
             let mut key: &'static str = param.into();
             key = key.strip_prefix(remove_prefix).unwrap_or(key);
             if let Some(value) = self.get(*param) {
-                yaml.insert(key.into(), value.clone());
+                yaml.insert(
+                    key.into(),
+                    serde_yaml::to_value(value.clone())
+                        .expect("failed to convert parameter value to YAML"),
```

Comment:
Could we return an error here instead of panicking?
And maybe add a test for it as well. Not sure if it makes sense but usually worth thinking about when adding a new failure condition.

Reply:
I wasn't sure about this case as all parameter values should be serializable as YAML, so we don't ever expect this to fail, even under adverse inputs, as they would have to be specified in YAML as well.

I've added a comment for now explaining this and will add a test exercising this path. Do you think it still makes sense to pass an error above as another defence-in-depth?
I'm still trying to understand where is it reasonable to use `unwrap/expect` (as in `key.strip_prefix(remove_prefix).unwrap_or(key)`) and where should we propagate the error.

Reply:
Ok, in this case I'm leaving it as is. Thanks for explaining the general strategy!

Ah, good catch about `unwrap_or`, that makes sense.